hermeto/core/package_managers/pip/main.py

```diff
@@ -676,20 +676,15 @@ def _get_external_requirement_filepath(requirement: PipRequirement) -> Path:
             filename = Path(orig_url.path).name
             filepath = Path(urlparse.unquote(filename))
         else:
-            # e.g. external-pyarn/pyarn-external-sha256-deadbeef.tar.gz
-            filepath = Path(
-                f"external-{package}", f"{package}-external-{algorithm}-{digest}{file_ext}"
-            )
+            # e.g. pyarn-external-sha256-deadbeef.tar.gz
+            filepath = Path(f"{package}-external-{algorithm}-{digest}{file_ext}")
```

Comment:
You put the `external-` fragment back in with the partial revert :) . Digest on its own should provide the required uniqueness (which was an oversight on my side during the initial review), and since we only need it for uniqueness, we IMO don't need to track the algorithm, but that's optional, `external` should go away regardless.